File: majorProject/src/main/java/com/aocc/framework/FileIO.java
package com.aocc.framework;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import android.content.SharedPreferences;

//THE FOLLOWING CODE IS SOURCED FROM THE KILOBOLT ANDROID FRAMEWORK

public interface FileIO {
    public InputStream readFile(String file) throws IOException;

    public OutputStream writeFile(String file) throws IOException;
    
    public InputStream readAsset(String file) throws IOException;
    
    public SharedPreferences getSharedPref();
}

File: majorProject/src/main/kotlin/com/aocc/framework/FileIO.kt
package com.aocc.framework

import android.content.SharedPreferences
import java.io.IOException
import java.io.InputStream
import java.io.OutputStream

//THE FOLLOWING CODE IS SOURCED FROM THE KILOBOLT ANDROID FRAMEWORK

interface FileIO {
    @Throws(IOException::class)
    fun readFile(file: String): InputStream

    @Throws(IOException::class)
    fun writeFile(file: String): OutputStream

    @Throws(IOException::class)
    fun readAsset(file: String): InputStream

    fun getSharedPref(): SharedPreferences
}
